### ISSUE TITLE
Depending on inside a TYPEDEF used more than once is ambiguous

### ISSUE THREAD

**Issue by smedilol** [TypeCobolTeam/TypeCobol#697]
**CHILD OF #663**

Clause `DEPENDING ON` refers to another variable in the source.
If a `TYPEDEF` contains a clause `DEPENDING ON` and at least 2 variables use this type, then the `DEPENDING ON` clause is ambiguous.

Example:
```cobol
       01 Account TYPEDEF STRICT.
           05 HolderNumber PIC 9(04) COMP-5.
           05 Holders      pic X(30)
                           OCCURS 99 TIMES
                           DEPENDING ON HolderNumber
                           INDEXED BY Idx.
       01 Account1 type Account.
       01 Account2 type Account.
```
This cause the error:
`Syntax error : Ambiguous reference to symbol HolderNumber`

**Comment by smedilol** [comment]
See #700

**Comment by collarbe** [comment]
What if we've got something like this : 

```cobol
       01 Account TYPEDEF STRICT.
           05 HolderNumber PIC 9(04) COMP-5.
           05 Holders      pic X(30)
                           OCCURS 99 TIMES
                           DEPENDING ON HolderNumber
                           INDEXED BY Idx.
       01 Account1 type Account.
       01 Account2 type Account.
       01 HolderNumber PIC 9(04) COMP-5.
```

Do we have to only consider the variable declared in the TypeDef or do we have to detect that HolderNumber is present multiple times and then throw a semantic error ? 

I think that we should only consider the variable declared in the TypeDef.

**Comment by smedilol** [comment]
> Do we have to only consider the variable declared in the TypeDef 

-> Yes, you only need to check HolderNumber in the TYPEDEF

**Comment by smedilol** [comment]
This code should raise an error:
```cobol
       01 MyType TYPEDEF STRICT.
         05 Count2 pic 9(05) comp-5.

         05 MyTable PIC X(1) occurs 99 
     *KO  var1 doesn't exists in this type
            depending on var1::Count2.

       01 var1 type MyType.
```

**Comment by collarbe** [comment]
```cobol
01 FirstType TYPEDEF STRICT.
             05 FirstCounter pic 9(05) comp-5.

01 MyType TYPEDEF STRICT.
         05 IncludedType TYPE FirstType.
         05 MyTable PIC X(1) occurs 99 DEPENDING ON FirstCounter .
         05 MyOtherTable PIC X(1) occurs 99 DEPENDING ON IncludedType::FirstCounter .
```
Is this possible ?? 😟

**Comment by smedilol** [comment]
Yes it should be possible but it's not urgent.
I'll create another issue for that.
